src/modules/mainMenu/utils.ts

```diff
@@ -0,0 +1,34 @@
+import cloneDeep from 'clone-deep'
+
+import { MAIN_MENU } from './constants/mainMenu'
+import { DropDownItem, MainMenuItemId, MenuItemKind, MenuTreeItem } from './types'
+
+import { Routes } from '../../constants/routes'
+
+const ADVANCED_ORDERS_MENU_TITLE = 'Advanced orders'
+
+export type BuildMainMenuTreeItemsParams = {
+  isAdvancedOrdersEnabled: boolean
+}
+
+export function buildMainMenuTreeItems({ isAdvancedOrdersEnabled }: BuildMainMenuTreeItemsParams): MenuTreeItem[] {
+  if (!isAdvancedOrdersEnabled) {
+    return MAIN_MENU
+  }
+
+  // Make a deep copy to avoid mutating original
+  const mainMenuCopy = cloneDeep(MAIN_MENU)
```

Comment:
```
const mainMenuCopy = [...MAIN_MENU]
```

Isn't it enough?

Reply:
Unfortunately not.
The list items are recreated but the inner references still shared.
I tried to cheat it like that but it didn't work. The error Elena reported was because of that.